src/contracts/NFTFactory.ts

```diff
@@ -70,7 +74,10 @@ export class NftFactory extends SmartContractWithAddress {
     try {
       const tx = await sendTx(
         estGas,
-        this.signer,
+        this.config.confidentialEVM === true &&
```

Comment:
Are we sure we have the proper` this.config`  here? (Same for all other similar classes, where this change was made) . I think the `Config` object is only available if we passed it on the constructor, or if we provided a network on the constructor.
This is what we have on the `SmartContract` constructor:
```
constructor(
    signer: Signer,
    network?: string | number,
    config?: Config,
    abi?: AbiItem[]
  ) {
    this.signer = signer
    this.config = config || new ConfigHelper().getConfig(network || 'unknown')
    this.abi = abi || this.getDefaultAbi()
  }
```
If we don't supply a `Config` object or a valid `network` it executes the the default:
`config = new ConfigHelper().getConfig('unknown')`
This will return a `null` config

Reply:
I agree, the easiest way is to make network mandatory and to have a check for supported networks.
What do you think @paulo-ocean?

Reply:
Actually, signer has a method called "signer.getChainId()", we can use that and have a check for supported networks.

Reply:
It will generate breaking change within SDK contracts classes and it implies a lor of modifications around all the classes making network a mandatory parameter. I already modified some tests to provide the network as well, but I think here comes the user responsability to set a config object with the necessary parameters. What I can improve to the check is to validate if config object has confidential EVM key or not, otherwise, use the regular signer.